src/nvim/api/win_config.c

```diff
@@ -236,18 +341,203 @@ void nvim_win_set_config(Window window, Dict(float_config) *config, Error *err)
   if (!win) {
     return;
   }
-  bool new_float = !win->w_floating;
+  tabpage_T *win_tp = win_find_tabpage(win);
+  bool was_normal = !win->w_floating;
+  bool has_split = HAS_KEY(config, float_config, split);
+  bool has_vertical = HAS_KEY(config, float_config, vertical);
   // reuse old values, if not overridden
-  FloatConfig fconfig = new_float ? FLOAT_CONFIG_INIT : win->w_float_config;
+  FloatConfig fconfig = win->w_float_config;
 
-  if (!parse_float_config(config, &fconfig, !new_float, false, err)) {
+  bool is_split = (!HAS_KEY(config, float_config, relative) || striequal(config->relative.data, ""))
```

Comment:
`is_split` is a bit ambiguous. I think `to_split` is better.

Reply:
I agree, to_split makes more sense.